lesson12/index.js

```diff
@@ -0,0 +1,75 @@
+const contents = document.getElementById('js-contents');
+const button = document.createElement('button');
+
+const displayLoadingImage = () => {
+    const loadingImage = document.createElement('img');
+    loadingImage.id = 'loadingImageId';
+    loadingImage.src = './loading-circle.gif';
+    contents.appendChild(loadingImage);
+}
+
+const removeLoadingImage = () => {
+    const loadingImage = document.getElementById('loadingImageId');
+    loadingImage.remove();
+}
+
+const getData = async (url) => {
+    try {
+        const response = await fetch(url);
+        return await response.json();
+    } catch (error) {
+        throw new Error(error);
+    }
+};
+
+const init = async () => {
+    displayLoadingImage();
+    removeButton();
+    try {
+        const { data } = await getData("https://myjson.dit.upm.es/api/bins/edwr");
+        createList(data);
+    } catch (e) {
+        errorText(`エラーが発生しました: 詳細 ${e}`);
+    }
```

Comment:
IMO
I forgot to share what I learned from Morita-san.

>  - When the data is returned, evaluate if the data content is in there. If not, say "data is empty".
>  - If it is there, run createList.

```suggestion
 try {
    const { data } = await getData("https://myjson.dit.upm.es/api/bins/edwr");
    if (data.length === 0) {
      errorText(`エラーが発生しました: 詳細 data is empty`);
      return;
    }
    createList(data);
  } catch (e) {
    errorText(`エラーが発生しました: 詳細 ${e}`);
  }
```


To verify when data is empty
```js
 try {
    const { data } = await getData("https://myjson.dit.upm.es/api/bins/edwr");
    if ([].length === 0) {
      errorText(`エラーが発生しました: 詳細 data is empty`);
      return;
    }
    createList(data);
  } catch (e) {
    errorText(`エラーが発生しました: 詳細 ${e}`);
  }
```
Since length is 0, data is empty will be displayed.
You can try it if you like.

Reply:
Thank you for your reviewing!
I added a condition when the data is empty.
-> 0dbce59